python/ray/data/dataset.py

```diff
@@ -807,12 +840,16 @@ def drop_columns(
                 ray (e.g., num_gpus=1 to request GPUs for the map tasks).
         """  # noqa: E501
 
+        # Historically, we have also accepted lists with duplicate column names.
+        # This is not tolerated by the underlying pyarrow.Table.drop_columns method.
+        cols_without_duplicates = list(set(cols))
+
```

Comment:
i think we should just enforce this via validation / raise an error

Reply:
This is a breaking change then!
Still?

Reply:
I think it's fine, yes.